konveyor/tackle-pathfinder
# TACKLE-9 : questionnaire reworded

Issue : https://github.com/konveyor/tackle-pathfinder/issues/59

Features included :
* Categories, Questions and Options reworded
* Added 1 Hybrid cloud Option

Rewording file : 
[Pathfinder Assessment Questions and suggestions.ods](https://github.com/konveyor/tackle-pathfinder/files/6541056/Pathfinder.Assessment.Questions.and.suggestions.ods)

Test Case : 
* using the UI
  * create a new Assessment and check the Categories, Questions and Options against the rewording file
 
* using CURL
  * utilities needed : `curl` and `jq`
  * cd to the project folder 
  * start  keycloak
  ```Shell
  podman run --name keycloak --rm \
            -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin \
            -e KEYCLOAK_IMPORT=/tmp/keycloak/quarkus-realm.json \
            -e DB_VENDOR=h2 -p 8180:8080 -p 8543:8443 -v ./src/main/resources/keycloak:/tmp/keycloak:Z \
            jboss/keycloak:12.0.2
  ```
  * start postgresql
  ```Shell
  podman run -it \                 
            --name petclinic -e POSTGRES_USER=postgres \
            -e POSTGRES_PASSWORD=petclinic -e POSTGRES_DB=petclinic \
            -p 5433:5432 postgres:10.6
  ```
  * execute in dev mode
  ```Shell
  ./mvnw clean quarkus:dev
  ```
  * get token
  ```Shell
  export access_token=$(\
    curl -X POST http://localhost:8180/auth/realms/quarkus/protocol/openid-connect/token \
    --user backend-service:secret \
    -H 'content-type: application/x-www-form-urlencoded' \
    -d 'username=alice&password=alice&grant_type=password' | jq --raw-output '.access_token' \
    )
   ```
  * create assessment
  ```Shell
  curl "http://localhost:8085/pathfinder/assessments" \                                         
            -H 'Content-Type: application/json' -H 'Accept: application/json' -H "Authorization: Bearer $access_token" \
            -d '{ "applicationId": 340 }' -w "%{http_code}"
  ```
  * Expect a similar JSON to this
  ```Shell
  {"id":629,"applicationId":340,"status":"STARTED"}201
  ```
  * get assessment
  ```Shell
  curl -X GET "http://localhost:8085/pathfinder/assessments/629" -H 'Accept: application/json' \
            -H "Authorization: Bearer $access_token" -v | jq . 
  ```
  * compare the returned JSON against the rewording file
